src/components/cafes/[id]/FlavorItem/index.tsx

```diff
@@ -1,41 +1,39 @@
-import  HashTag  from '@/components/common/HashTag';
+import HashTag from '@/components/common/HashTag';
 import { dot, flavorItem, flavorList } from './FlavorItem.css';
-export interface FlavorListProps {
-  fruity: string;
-  floral: string;
-  sweet: string;
-  spices: string;
-  roasted: string;
-  green: string;
-  sour: string;
-  nutty: string;
-  other: string;
-}
+import { FLAVOR_DATA } from '@/constants/flavor';
+
+type FlavorCategory = keyof typeof FLAVOR_DATA | 'other';
+
+const getFlavor = (name: string): FlavorCategory => {
+  // 카테고리 이름이나 아이템 중 일치하는 것 찾기
+  //1.[fruity, {name:'과일',color:...,itmes:['사과','딸기'...]}]형태로 변환
+  //2. 1번째 요소인 {name:'과일',color:...,items:['사과',...]}의 items 배열의 접근하여 name이 있는지 확인!
+  //3. name이 있으면 name을 가지고있는 0번째 요소인 Key값 반환! 아니면 'other' color로 반환!
+  const matchedCategory = Object.entries(FLAVOR_DATA).find(([_, flavorValue]) => {
+    return flavorValue.items.includes(name);
+  });
```

Comment:
key 값을 활용할 게 아니라면 아래처럼 values 메서드를 사용하면 될 것 같아요~
```suggestion
  const matchedCategory = Object.values(FLAVOR_DATA).find((flavorValue) => {
    return flavorValue.items.includes(name);
  });
```